.github/workflows/deploy-new-version.yml

```diff
@@ -0,0 +1,116 @@
+name: Deploy New Version
+run-name: New deployment for "${{ github.ref_name }}" triggered by ${{ github.actor }}
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+  push:
+    branches:
+      - main
+
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Enter the version number'
+        required: true
+        default: 'latest'
+
+jobs:
+  format_and_lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '16.14.2'
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Run ESLint
+        run: npm run lint
+
+      - name: Run Prettier
+        run: npm run prettier
+
+  check-version:
+    runs-on: ubuntu-latest
+    needs: format_and_lint
+
+    steps:
+      - name: Should continue
+        run: |
+          if [[ ${{ github.ref_name != 'main' && inputs.version == '' }} ]]; then
```

Comment:
Why like this instead of `if` on the job? https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/using-conditions-to-control-job-execution#example-only-run-job-for-specific-repository